src/libraries/Microsoft.PowerFx.Core/Syntax/Nodes/TypeLiteralNode.cs

```diff
@@ -113,6 +114,28 @@ public override void PostVisit(TableNode node)
             {
             }
 
+            public override bool PreVisit(CallNode node)
+            {
+                if (ValidRecordOfNode(node))
+                {
+                    return true;
+                }
+
+                _errors.Add(new TexlError(node, DocumentErrorSeverity.Severe, TexlStrings.ErrTypeLiteral_InvalidTypeDefinition, node.ToString()));
+                return false;
+            }
+
+            public override bool PreVisit(ListNode node)
+            {
+                if (node.Parent != null && node.Parent is CallNode cn && ValidRecordOfNode(cn))
```

Comment:
>node.Parent != null &&  [](http://example.com/codeflow?start=20&length=23)

Is the `!= null` necessary? I think `node.Parent is CallNode cn` will not return true if node.Parent is null